src/sycl/per_tensor_quant_fp8.cpp

```diff
@@ -0,0 +1,235 @@
+        output_s.data_ptr<float>(),                                                     \
+        num_elements);                                                                  \
+    sycl_kernel_submit(global_range, local_range, Q, kernel);                           \
+    Q.wait_and_throw();                                                                 \
+  } while (0)
+
+#define DISPATCH_VEC_SIZE(T, DST_DTYPE, vec_size)                                       \
+  switch (vec_size) {                                                                   \
+    case 1:                                                                             \
+      LAUNCH_KERNEL(T, DST_DTYPE, 1);                                                   \
+      break;                                                                            \
+    case 2:                                                                             \
+      LAUNCH_KERNEL(T, DST_DTYPE, 2);                                                   \
+      break;                                                                            \
+    case 4:                                                                             \
+      LAUNCH_KERNEL(T, DST_DTYPE, 4);                                                   \
+      break;                                                                            \
+    case 8:                                                                             \
+      LAUNCH_KERNEL(T, DST_DTYPE, 8);                                                   \
+      break;                                                                            \
+    case 16:                                                                            \
+      LAUNCH_KERNEL(T, DST_DTYPE, 16);                                                  \
+      break;                                                                            \
+    default:                                                                            \
+      throw std::runtime_error("Unsupported vector size: " + std::to_string(vec_size)); \
+  }
+
+  // Dispatch based on input and output types
+  if (input.scalar_type() == at::ScalarType::Half) {
+    if (output_q.scalar_type() == at::ScalarType::Float8_e4m3fn) {
+      int vec_size = preferred_vector_width(dpcppGetDeviceIdOfCurrentQueue(), sizeof(sycl::half));
+      DISPATCH_VEC_SIZE(sycl::half, cutlass::float_e4m3_t, vec_size);
+    }
+  } else if (input.scalar_type() == at::ScalarType::BFloat16) {
+    if (output_q.scalar_type() == at::ScalarType::Float8_e4m3fn) {
+      int vec_size = preferred_vector_width(dpcppGetDeviceIdOfCurrentQueue(), sizeof(sycl::ext::oneapi::bfloat16));
+      DISPATCH_VEC_SIZE(sycl::ext::oneapi::bfloat16, cutlass::float_e4m3_t, vec_size);
+    }
+  } else {
+    throw std::runtime_error("Unsupported data type for per-tensor FP8 quantization");
+  }
```

Comment:
use ` AT_DISPATCH_REDUCED_FLOATING_TYPES`

Reply:
once you used  AT_DISPATCH_REDUCED_FLOATING_TYPES, you can just put `DISPATCH_VEC_SIZE` inside the function handler, since you will need to call it only once, no need to use MACRO.